File: sdk/kotlin/src/test/kotlin/com/loomup/client/SubKeyTest.kt
package com.loomup.client

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class SubKeyTest {
    @Test
    fun parseSubKeySplitsOnlyOnFirstHash() {
        val a = parseSubKey("todos")
        assertEquals("todos", a.first)
        assertNull(a.second)

        val b = parseSubKey("todos#1")
        assertEquals("todos", b.first)
        assertEquals("1", b.second)

        val c = parseSubKey("todos#a#b#c")
        assertEquals("todos", c.first)
        assertEquals("a#b#c", c.second)

        assertEquals("todos#a#b", makeSubKey("todos", "a#b"))
        val round = parseSubKey(makeSubKey("notes", "x#y"))
        assertEquals("notes", round.first)
        assertEquals("x#y", round.second)
    }
}

File: sdk/flutter/test/sub_key_test.dart
import 'package:loomup/loomup.dart';
import 'package:test/test.dart';

void main() {
  test('parseSubKey splits only on first hash', () {
    final a = parseSubKey('todos');
    expect(a.table, 'todos');
    expect(a.rowId, isNull);

    final b = parseSubKey('todos#1');
    expect(b.table, 'todos');
    expect(b.rowId, '1');

    final c = parseSubKey('todos#a#b#c');
    expect(c.table, 'todos');
    expect(c.rowId, 'a#b#c');

    expect(makeSubKey('todos', 'a#b'), 'todos#a#b');
    final round = parseSubKey(makeSubKey('notes', 'x#y'));
    expect(round.table, 'notes');
    expect(round.rowId, 'x#y');
  });
}
